app/src/main/java/io/appium/uiautomator2/model/UiElementSnapshot.java

```diff
@@ -158,11 +164,49 @@ private static void putAttribute(Map<Attribute, Object> attribs, Attribute key,
             case TEXT:
                 return AxNodeInfoHelper.getText(node, true);
             case HINT:
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                    return node.getHintText();
-                } else {
-                    return null;
+                return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O ? node.getHintText() : null;
+            case IMPORTANT_FOR_ACCESSIBILITY:
+                return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? node.isImportantForAccessibility() : null;
+            case SCREEN_READER_FOCUSABLE:
+                return Build.VERSION.SDK_INT >= Build.VERSION_CODES.P ? node.isScreenReaderFocusable() : null;
+            case INPUT_TYPE:
+                return node.getInputType() != 0 ? node.getInputType() : null;
+            case DRAWING_ORDER:
+                return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? node.getDrawingOrder() : null;
+            case SHOWING_HINT_TEXT:
+                return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O ? node.isShowingHintText() : null;
+            case ACTIONS:
+                StringBuilder actionsBuilder = new StringBuilder();
```

Comment:
consider moving this block of code (compiling actions to an attribute string) to a separate helper method